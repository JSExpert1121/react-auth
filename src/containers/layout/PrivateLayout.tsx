/**
 * @name PrivateLayout
 * @description Common Layout for private pages. 
 * @version 0.1 (only skeleton)
 * @version 0.2 (basic)
 * @todo Real layout
 * @author Darko
 */
import React from 'react';

// material ui components & icons
import Box from '@material-ui/core/Box';
import { makeStyles, Theme } from '@material-ui/core/styles';

// custom components
import Header from './PrivateHeader';
import Footer from './PrivateFooter';


const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: '#FFF',
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        height: '100%'
    },
    contents: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        flex: 1
    }
}));


interface IPrivateLayoutProps {
}

export const PrivateLayout: React.FC<IPrivateLayoutProps> = (props) => {
    const { children } = props;
    const classes = useStyles({});

    return (
        <Box className={classes.root}>
            <Header />
            <Box component='main' className={classes.contents}>
                {children}
            </Box>
            <Footer />
        </Box>
    )
};

export default PrivateLayout;
