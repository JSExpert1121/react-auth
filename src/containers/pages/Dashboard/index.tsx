import * as React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router-dom';

// material ui components & icons
import Box from '@material-ui/core/Box';
import { makeStyles, Theme, StyledComponentProps } from '@material-ui/core/styles';

// 3rd party libraries
import { injectIntl, WrappedComponentProps } from 'react-intl';
import messages from 'translations/messages/page.dashboard';

// actions & apis & types
import * as GlobalActions from 'store/actions/global';
// import * as UserActions from 'store/actions/user';
// import * as MemberApi from 'service/member';


const useStyles = makeStyles((theme: Theme) => ({
    root: {
        display: 'flex',
        width: '100%',
    }
}));

interface IDashboardProps extends RouteComponentProps, StyledComponentProps, WrappedComponentProps {
    setTitle: (title: string) => void;
}

const Dashboard: React.FunctionComponent<IDashboardProps> = (props) => {

    const { setTitle, intl } = props;
    React.useEffect(() => {
        setTitle(intl.formatMessage(messages.title));
    });

    const classes = useStyles({});
    return (
        <Box className={classes.root}>
            Dashboard
        </Box>
    );
};

const mapStateToProps = (state: any) => ({
});

const mapDispatchToProps = (dispatch: any) => ({
    setTitle: (title: string) => dispatch(GlobalActions.setTitle(title)),
});

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(Dashboard));
