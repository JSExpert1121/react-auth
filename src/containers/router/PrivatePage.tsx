import React from 'react';
import { Route, RouteComponentProps, Switch, Redirect } from 'react-router';
import CssBaseline from '@material-ui/core/CssBaseline';

import PrivateLayout from 'containers/layout/PrivateLayout';

// custom pages
import Dashboard from 'containers/pages/Dashboard';
import AccountPage from 'containers/pages/Account';

interface IPrivatePageProps extends RouteComponentProps {

}

const PrivatePage: React.FC<IPrivatePageProps> = (props) => {
    const { match } = props;
    return (
        <PrivateLayout {...props}>
            <CssBaseline />
            <Switch>
                <Route path={`${match.url}/dashboard`} component={Dashboard} />
                <Route path={`${match.url}/account`} component={AccountPage} />
                <Redirect to={`${match.url}/dashboard`} />
            </Switch>
        </PrivateLayout>
    )
}

export default PrivatePage;