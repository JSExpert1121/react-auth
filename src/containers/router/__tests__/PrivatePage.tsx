import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createShallow } from '@material-ui/core/test-utils';
import PrivatePage from '../PrivatePage';
import PrivateLayout from 'containers/layout/PrivateLayout';
import Dashboard from 'containers/pages/Dashboard';
import AccountPage from 'containers/pages/Account';


configure({ adapter: new Adapter() });

describe('Public Page', () => {

    const shallow = createShallow();
    const wrapper = shallow(<PrivatePage match={{ url: '/user', path: '/user', isExact: false, params: {} }} />);

    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('Public Layout', () => {
        expect(wrapper.find(PrivateLayout)).toHaveLength(1);

        expect(wrapper.find(Switch)).toHaveLength(1);
        const children = wrapper.find(Switch).props().children;
        expect(children).toHaveLength(3);

        expect(children[0].type).toEqual(Route);
        expect(children[0].props).toEqual({ path: '/user/dashboard', component: Dashboard });
        expect(children[1].type).toEqual(Route);
        expect(children[1].props).toEqual({ path: '/user/account', component: AccountPage });
        expect(children[2].type).toEqual(Redirect);
        expect(children[2].props).toEqual({ to: '/user/dashboard' });
    })
});