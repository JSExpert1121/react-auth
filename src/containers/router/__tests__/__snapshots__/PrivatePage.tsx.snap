// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Public Page snapshot 1`] = `ShallowWrapper {}`;
